src/suggestion.c

```diff
@@ -0,0 +1,448 @@
+/*
+ * A series of functions to generate spelling suggestions
+ * 
+ * See suggestion.h for function documentation
+ */
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <assert.h>
+#include <unistd.h>
+#include <string.h>
+#include "suggestion.h"
+#include "trie.h"
+
+int has_children(trie_t *t, char *s) {
+    
+    return !trie_search(t, s);
+}
+
+/* See suggestion.h */
+int cmp_match(const void* a, const void* b) {
+
+    // Recast a and b as match_t's
+    match_t *aa = *(match_t **)a;
+    match_t *bb = *(match_t **)b;
+
+    // Put null entries at the end of the list
+    if (aa == NULL) {
+        return 1;
+    } else if (bb == NULL) {
+        return -1;
+    }
+
+    // Prioritize matches with the higher edits_left score
+    if (aa->edits_left != bb->edits_left) {
+        return bb->edits_left - aa->edits_left;
+    }
+
+    // Otherwise sort lexographically
+    return strcmp(aa->str,bb->str);
```

Comment:
Let's use `strncmp` here against some compile-time constant, and let's also stick a space after that comma.

Reply:
changed to strncmp